    @Test
    @RunTestInLooperThread
    public void trustedRootCA_syncShouldFailWithoutTrustedCA() throws InterruptedException {
        String username = UUID.randomUUID().toString();
        String password = "password";
        SyncUser user = SyncUser.logIn(SyncCredentials.usernamePassword(username, password, true), Constants.AUTH_URL);

        // 1. Copy a valid Realm to the server
        //noinspection unchecked
        final SyncConfiguration syncConfig = configurationFactory.createSyncConfigurationBuilder(user, Constants.USER_REALM)
                .schema(StringOnly.class)
                .build();
        Realm realm = Realm.getInstance(syncConfig);

        realm.beginTransaction();
        realm.createObject(StringOnly.class).setChars("Foo");
        realm.commitTransaction();

        // make sure the changes gets to the server
        SyncManager.getSession(syncConfig).uploadAllLocalChanges();
        realm.close();
        user.logOut();

        // 2. Local state should now be completely reset. Open the Realm again with a new configuration which should
        // download the uploaded changes.
        user = SyncUser.logIn(SyncCredentials.usernamePassword(username, password), Constants.AUTH_URL);
        //noinspection unchecked
        SyncConfiguration syncConfigSSL = configurationFactory.createSyncConfigurationBuilder(user, Constants.USER_REALM_SECURE)
                .name("useSsl")
                .schema(StringOnly.class)
                .trustedRootCA("untrusted_ca.pem")
                .build();
        // waitForInitialRemoteData will throw an Internal error (125): Operation Canceled
        SystemClock.sleep(TimeUnit.SECONDS.toMillis(2));
        realm = Realm.getInstance(syncConfigSSL);
        try {
            assertTrue(realm.isEmpty());
        } finally {
            realm.close();
        }
        looperThread.testComplete();
    }

